dspace-api/src/test/java/org/dspace/discovery/MockSolrServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.stereotype.Service;
+
+/**
+ * Mock SOLR service for the Search Core
+ */
+@Service
+public class MockSolrServiceImpl extends SolrServiceImpl implements InitializingBean {
```

Comment:
The (pre-existing) testing framework builds a test dspace install directory by combining all "src/test/data/dspaceFolder" directories of all modules (see https://github.com/DSpace/DSpace/blob/dspace-6.2/src/main/assembly/testEnvironment.xml#L30). So all Spring beans defined in test-specific Spring XML files, need to be present in the test resources of all modules. That is why I had to add a copy of that class to `dspace-api` and `dspace-spring-rest`. Another example of such a class is `org.dspace.identifier.MockDOIConnector`.

However, since we're not doing any SOLR based testing in dspace-api (yet), I've disabled the embedded SOLR server in those tests to not slow them down.